src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+                        if (Is25ManRaid()) // one wave each 10s. one wave=1 zombie in 10man and 2 zombies in 25man.
+                        {
+                            DoSummon(NPC_ZOMBIE, PosSummon[rand32() % 3]);
+                            DoSummon(NPC_ZOMBIE, PosSummon[rand32() % 3]);
+                        }
+                        else
+                            DoSummon(NPC_ZOMBIE, PosSummon[0]);
+                        events.ScheduleEvent(EVENT_SUMMON, 10 * IN_MILLISECONDS);
+                        break;
+                    case EVENT_SEARCH_ZOMBIE_SINGLE:
+                    {
+                        Creature* zombie = nullptr;
+                        for (SummonList::const_iterator itr = summons.begin(); !zombie && itr != summons.end(); ++itr)
+                            if (ObjectAccessor::GetCreature(*me, *itr)->IsWithinDistInMap(me, 10.0))
+                                zombie = ObjectAccessor::GetCreature(*me, *itr);
+                        if (zombie)
+                        {
+                            zombieToBeEatenGUID = zombie->GetGUID(); // save for later use
+                            Talk(EMOTE_SPOTS_ONE);
+
+                            // the soon-to-be-eaten zombie should stop moving and stop attacking
+                            zombie->AI()->SetData(1, STATE_ZOMBIE_TOBE_EATEN); // change the state of the zombie to STATE_ZOMBIE_TOBE_EATEN
+
+                            // gluth should turn toward the zombie and stop AAs on his primary target. He then pauses for a few seconds
+                            me->GetMotionMaster()->MoveFollow(zombie, 0, 0);
+                            me->SetReactState(ReactStates::REACT_PASSIVE);
+                            me->SetFacingToObject(zombie);
+                            me->SetTarget(ObjectGuid::Empty);
+                            state = STATE_GLUTH_EATING;
+
+                            events.ScheduleEvent(EVENT_KILL_ZOMBIE_SINGLE, 2.5 * IN_MILLISECONDS);
+                        }
+
+                        events.ScheduleEvent(EVENT_SEARCH_ZOMBIE_SINGLE, 7 * IN_MILLISECONDS);
                         break;
+                    }
+                    case EVENT_KILL_ZOMBIE_SINGLE:
+                    {
+                        Creature* zombieToBeEaten = ObjectAccessor::GetCreature(*me, zombieToBeEatenGUID);
+                        if (zombieToBeEaten && zombieToBeEaten->IsAlive() && zombieToBeEaten->IsWithinDistInMap(me, 10.0))
+                            DoCast(zombieToBeEaten, SPELL_ZOMBIE_CHOW_SEARCH_SINGLE); // do the killing + healing in done inside by spell script see below
+
+                        zombieToBeEatenGUID = ObjectGuid::Empty;
+                        state = STATE_GLUTH_NORMAL;
+
+                        // and then return on primary target
+                        me->SetReactState(ReactStates::REACT_AGGRESSIVE);
+                        me->SetFacingToObject(nullptr);
```

Comment:
This line is a bit akward. I'm not really sure how it works. I wanted to null the effect of line 201 since I noticed when Gluth is stunned (by the effect of Decimate), he would turn toward the zombie he previously killed. But it was a bug in an older version. I'm not sure it's still there. And I'm not sure if this line do anything anyway.